lib/zendesk_api/middleware/response/sanitise_response.rb

```diff
@@ -0,0 +1,13 @@
+module ZendeskAPI
+  module Middleware
+    module Response
+      class SanitiseResponse < Faraday::Response::Middleware
+        def on_complete(env)
+          unless env[:body].strip.empty?
+            env[:body] = env[:body].encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '')
```

Comment:
we can make the gemspec say >= 2.1 to be safe / have nobody use a newer
version on an old project ... but pushing users to upgrade their ruby is
always good :)

On Mon, Sep 7, 2015 at 10:00 AM, Gary Grossman <notifications@github.com>
wrote:

> In lib/zendesk_api/middleware/response/sanitise_response.rb
> <https://github.com/zendesk/zendesk_api_client_rb/pull/266#discussion_r38874031>
> :
>
> > @@ -0,0 +1,13 @@
> > +module ZendeskAPI
> > +  module Middleware
> > +    module Response
> > +      class SanitiseResponse < Faraday::Response::Middleware
> > +        def on_complete(env)
> > +          unless env[:body].strip.empty?
> > +            env[:body] = env[:body].encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '')
>
> I was basically going off the .travis.yml in this repo which still goes
> back to 1.9.3... you think we can safely drop support for pre-2.1?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/zendesk_api_client_rb/pull/266/files#r38874031>
> .
>